transport/grpc/client.go

```diff
@@ -141,17 +141,33 @@ type clientOptions struct {
 	printDiscoveryDebugLog bool
 }
 
+// NewClient returns a GRPC connection.
+func NewClient(opts ...ClientOption) (*grpc.ClientConn, error) {
+	options, grpcOpts := makeOptions(false, opts...)
+	return newClient(options, grpcOpts)
+}
+
+// NewClientInsecure returns an insecure GRPC connection.
+func NewClientInsecure(opts ...ClientOption) (*grpc.ClientConn, error) {
+	options, grpcOpts := makeOptions(true, opts...)
+	return newClient(options, grpcOpts)
+}
+
 // Dial returns a GRPC connection.
+// Deprecated: use NewClient instead.
 func Dial(ctx context.Context, opts ...ClientOption) (*grpc.ClientConn, error) {
-	return dial(ctx, false, opts...)
+	options, grpcOpts := makeOptions(false, opts...)
+	return dial(ctx, options, grpcOpts)
```

Comment:
`Dial` has been deprecated. Can use `NewClient` directly? For example: `return NewClient(opts...)`

Reply:
The `NewClient` function can be used directly; 

However, the original timeout and block connection logic supported by the gRPC `Dial` will be completely discarded, rendering them incompatible.

For the sake of compatibility, it's advisable to follow the same pattern as the grpc-go library
[Anti-Patterns of Client creation](https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md)
> For these reasons, using grpc.Dial is discouraged. Even though it is marked as deprecated, we will continue to support it until a v2 is released (and no plans for a v2 exist at the time this was written).

merely annotating the deprecated features. These can then be fully removed in a future major version update.


Ultimately, this decision hinges on the project maintainer's perspective. If deemed necessary, direct replacement without regard for the original logic's compatibility can be pursued. I will do this.